Frontend2

doxikus opened the pull request:
New theme for ESPeasy. If it is possible to have separated branch for testing purpose, few days then it can be merged to mega.  

Thanks

Commit:
git ignore

Commit:
Merge remote-tracking branch 'upstream/mega' into mega

Commit:
fix conflict

Commit:
Merge remote-tracking branch 'upstream/mega' into mega

Commit:
Merge remote-tracking branch 'upstream/mega' into mega

Commit:
Merge remote-tracking branch 'upstream/mega' into mega

Commit:
new web interface

uzi18 commented:
@doxikus screenshot please :)

doxikus commented:
Hi @uzi18 here it is, from my mobile phone

![screenshot_20180319-091201](https://user-images.githubusercontent.com/236233/37584643-9acaebf2-2b56-11e8-88b3-9653797c4930.png)
![screenshot_20180319-091220](https://user-images.githubusercontent.com/236233/37584644-9ae4c05e-2b56-11e8-8cce-5409aeeed941.png)
![screenshot_20180319-091229](https://user-images.githubusercontent.com/236233/37584645-9b0607be-2b56-11e8-9c02-6cbdc927faea.png)
![screenshot_20180319-091312](https://user-images.githubusercontent.com/236233/37584646-9b1fa71e-2b56-11e8-90ec-f09bf6cabd34.png)

soif commented:
Looks cool, but are you sure that your PR is complete : 
only 1 file (+ .gitignore) slightly modified :expressionless:

doxikus commented:
HI @soif only WebServer.ino is modified, updated markup and added separted css an js files. In feature with external css included I can faster resolve issues with theme. Theme is still rough but after some feedback I can adjust changes

TD-er commented:
Should we have some check like "if file exists on flash, use that, or else use embedded" ?

And nice to see the buttons on the Tools page now have equal width :)

doxikus commented:
HI, I think that there is already logic, first use embeded css and if there
is esp.css then use that. I would like to avoid that, write upon compiling
FW to flash css and js and remove from template css, split logic.

On Mon, Mar 19, 2018 at 10:58 AM, Gijs Noorlander <notifications@github.com>
wrote:

> Should we have some check like "if file exists on flash, use that, or else
> use embedded" ?
>
> And nice to see the buttons on the Tools page now have equal width :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/letscontrolit/ESPEasy/pull/1122#issuecomment-374158071>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOayQvu1CXlLna1fsFvc1QEIDWX_xKuks5tf4EsgaJpZM4SvrhA>
> .
>



-- 
Stanislav Mihić
Web developer
-------------------------------------------------------------------
E-mail: stanislav.mihic@gmail.com
-------------------------------------------------------------------
GSM :  0959046114
-------------------------------------------------------------------
WEB : www.veselistudio.com
----------------------------------------------------

soif commented:
@doxikus OK so your css is currently missing.

BTW you should avoid giving the same  Id  ('shrink'). to different page bodies, You rather should add it to the body class, and reserve the id to surcharge the css for specifics page when needed (because of the higher position of id in the CSS hierarchy)....

doxikus commented:
Yes you are right about body id, with id js is simpler, I can move down
trough hierarchy on header so body will be clean for feature use.

On Mon, Mar 19, 2018 at 12:01 PM, soif <notifications@github.com> wrote:

> @doxikus <https://github.com/doxikus> OK so your css is currently missing.
>
> BTW you should avoid giving the same Id ('shrink'). to different page
> bodies, You rather should add it to the body class, and reserve the id to
> surcharge the css for specifics page when needed (because of the higher
> position of id in the CSS hierarchy)....
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/letscontrolit/ESPEasy/pull/1122#issuecomment-374174710>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOayQUFj9YkMAd0rJBr8JsAbNFC3ma7ks5tf5ADgaJpZM4SvrhA>
> .
>



-- 
Stanislav Mihić
Web developer
-------------------------------------------------------------------
E-mail: stanislav.mihic@gmail.com
-------------------------------------------------------------------
GSM :  0959046114
-------------------------------------------------------------------
WEB : www.veselistudio.com
----------------------------------------------------

soif commented:
it would be more elegant, and more future proof IMHO, if you can 👍

happytm commented:
Good job. Please have a look at another library at [https://github.com/ayushsharma82/ESP-DASH](url) too for inspiration.

s0170071 commented:
@doxikus well done so far :-)  

I think I mentioned it before, but especially for mobile devices we should think about a redesign of the main page. A lot of useless info there blocking the space.  
What about the list of other ESPs contains just the name (which contains the link) and something else, e.g. the uptime and one (or several) select sensor reading(s). 
Example. 
```

SMA        1.2 days  4554W
Outdoor    45 days   12.3°C
Sunshine   1 day     12mcd
Livingroom 1 day     12°C
                     55%r.h.
```




I like the ESP-DASH. That and the links to the other ESPs in the network would be ideal.
 
Hey, what about a DASH plugin that allows to configure the main page ,e.g. what sensor data is displayed

Grovkillen commented:
![untitled_page](https://user-images.githubusercontent.com/26706274/37645639-bb28e94a-2c27-11e8-8b21-70d20722fc8d.png)
I'm thinking about something like this. ("1": these are not dropdown menus but rather folding divs). The main page is just the info page. On a desktop these would all be unfolded and in two columns, on phone the first "unit info" should be unfolded, the rest not.

Thoughts? :)

TD-er commented:
`Thoughts? :)`

Well there is a "report" option next to your message and I was tempted to press it... Comic sans in a design... ;)

But I like the idea of folding stuff.
It could even be done using JS and collected via JSON query, so it is possible to do live updates.

Grovkillen commented:
@TD-er this is a quick mockup... just to show how to organize things. NOT a suggested design :laughing:

ghtester commented:
From my perspective it would be great to have the first page customizable. I mean some devices could work just as switches, some other just as temperature sensors etc. So (for instance) I would like to have a possibility to configure just ON/OFF buttons at first page on "switch" devices and sensor values with descriptions on "sensor" devices (ideally with autorefresh). The Main page could be there by default but not mandatory. What do you think about it?

s0170071 commented:
What about the main page has three frames. 

1. head with common info like uptime and freemem
2. custom frame provided by a plugin
3. list of other devices.

I think it would allow for a nice transition to allow each plugin to contribute a string to the 2nd frame. There could be a new plugin state (returnMainPageString), enabled by a checkbox in the plugin settings.
All devices strings are concatenated and finally added to the 2nd frame.

ghtester commented:
Well, I would keep the first (custom) page as simple as possible dashboard, perhaps with an optional link (button) which could open the Main page only when needed (or if there's no custom first page ever). I usually don't need to display any details about ESP device when it does all I need (and thanks to all of you contributors, it usually does perfectly! :-) ). So most often I only need to read a sensors statuses or to control a switches. My idea is to have a possibility to open the ESP device's first page directly from any PC/laptop/tablet/mobile phone's web browser and quickly access the preferred functions or to display a sensors values permanently (with automatic refresh at intervals). But I have no fiction how the custom page could be "connected" to appropriate plugins...

TD-er commented:
We could add some lines like [switch#1], which resolves to the plugin with that name and sensorvalue.
Then the plugin should be queried to show something typical for that plugin.
For example a switch should add a switch action + button.
A sensor should show a dial (or whatever visualization) with a proper range and indicator whether it is good or not.
I think up-to 4 things could be shown on the start page to allow for a fully customizable first page.

doxikus commented:
Well for start I don't use web interface for monitoring, domoticz handle that, web is browsed only for first time configuration and that's it. On other hand someone use it daily and there is a problem. To achieve all desired functionality an have all features, device memory on disposal is very limited and we need to have only features that we really need. For example switches, it is great thing to have that on main page along with sensor reading data. Maybe to go in that direction, on main page to have switches, current sensor reading/state and list of other esp devices, even maybe to have only basic info and have linked to more detailed page, or device page.

Also any visualization involve using some sort of javascript framework and that is to heavy for esp.

There is posibility to have thatbut for different devices, with small memory space only basic interface but for higher memory devices more features enabled.

doxikus commented:
Also with some json data we can create separated "APP" which will pull all data from network and display it on more UI friendly dashboard. But this is already reserved for different controllers (domotiz, hassio etc)

s0170071 commented:
@TD-er The content has to come from the plugin. That is the only way that gives you enough flexibility what is added and how much. If the user enables "main page content" on every plugin- so be it. That content is collected from each plugin upon main page delivery. There is no need for a limit.

Another pro argument: you can implement this step by step, one plugin at a time.

TD-er commented:
The thing with Javascript is that it doesn't take resources from the ESP. You run it in the browser and the http-calls could even be simpler.
And lots of people are also using the ESPeasy modules as stand-alone, not connected to a Domoticz/OpenHAB/HomeAssistant.

@s0170071 I totally agree that should be 'asked' to the plugin (object oriented) and there is more a practical limit like "not being too crowded".
Something like "less is more".
Quite often putting some kind of limit like that will start people to think about what is really needed. Ofcourse we shouldn't make that limit too strict, but rather enforce some kind of logical limit may improve things :)

uzi18 commented:
It will be fully customisable if we agree for html file as mainpage template on spiffs. 
We could send header with some scripts to fill id with values from json.
I think it is already possible with /json request.
It will be too heavy for 1MB flashes but we can use external link there.

On the other hand if we add checkbox on every values in plugins it could be enough.
Table Like:
Pluginname valuename value

doxikus commented:
Hello guys, for now I see there is much important pull request in lack of time i cannot promise when new interface will be ready and it has to many edge cases. So for now this pull request can be put on old until other thing is resolved :-(

Grovkillen commented:
@doxikus What's the status on this? I've been adding some more stuff to the Webserver.ino yesterday and today. And Gijs is looking into some more this evening. Should we close this PR and you will get back later?

doxikus commented:
@Grovkillen hi, please close for now, i will later jump in. There is still some unanswered questions. It think in feature release we can talk about this topic. In my absence and lack of time best for now is to close this pull request.

Thanks
